packages/evm/src/interpreter.ts

```diff
@@ -918,6 +928,9 @@ export class Interpreter {
       Object.assign(this._result.selfdestruct, selfdestruct)
       // update stateRoot on current contract
       const account = await this._eei.getAccount(this._env.address)
+      if (!account) {
+        throw new Error('could not read contract account')
```

Comment:
Generally - yes - we can use to detect systemic issues early and not have this disguided by account creation (which then goes on with a wrong nonce or so e.g.). That said, I removed most of these patterns (throw) again in the storage cache PR, since these places tend to need some dedicated attention and the initial test failures were too overwhelming.

That often doesn't mean that it would be plain wrong to change, in fact at the end of my storage cache work I re-introduced some of these (also for finding my own bugs in the code).